Improve warnings for qnn runner

abhinaykukkadapu opened the pull request:
Summary:
Changes:
1. add `--seq_len` param to llama script to distinguish max_seq_len which is compile time param
2. Add warnings in the runner when `seq_len` is clamped to `max_seq_len` to avoid silently clamping it.
3. Add warnings in the token generator when EOS is not reached due to insufficient seq_len or max_seq_len.

Differential Revision: D86696759

## Tests

Use --seq_len=600, prompt_len=512

```
I 00:00:02.883890 executorch:token_generator.cpp:333] Warning: Generation stopped at seq_len limit (600) without reaching EOS token. Response may be incomplete.
I 00:00:02.884094 executorch:token_generator.cpp:346] - seq_len (600) is less than compiled max_seq_len (1024). Consider increasing --seq_len (up to 1024).
```

Use --seq_len=2048, prefill_ar_len=1024

```
I 00:00:00.546967 executorch:runner.cpp:385] Warning: Requested seq_len (2048) exceeds compiled max_seq_len (1024). Clamping to 1024.
```

winskuo-quic reviewed (commented):
Hi @abhinaykukkadapu,
Thanks for the PR.
I would like to know if we could achieve the same thing with the combination of 
`--pre_gen_pte & --max_seq_len`.

For example:
During compile time, you can provide:
`--max_seq_len 1024 --compile_only`

During inference, you can provide:
`--max_seq_len 512 --pre_gen_pte ./path_to_pregen_pte`

winskuo-quic commented:
> @winskuo-quic Thanks for the quick review, the goal of this additional param is to avoid confusing the users of the script thinking that --max_seq_len can be dynamic but it is a static param and is fixed during compilation.
> 
> Currently, one can pass `--max_seq_len` for inference which actually made me think that total context length can be changed dynamically and i only found out after digging through the code that is is piped as `seq_len` to the runner. With this new param, we have a clear distinction, that `--max_seq_len` is used for compile time and `--seq_len` is during runtime/ inference. Functionally this change is a no-op and it only improves user experience and making it visible to the user when we clamp it down.

I see.
I think it makes sense to add some warning messages in `.cpp` files to guide users, which is helpful.
However, for `llama.py`, I would like to know if you think flag `--max_seq_len` is misleading? The `--max_seq_len` flag can actually be set to a different number every time during execution, as long as it is shorter than the `--max_seq_len` used during compilation.

abhinaykukkadapu committed:
Introduce seq_len as inference param, and improve warnings (#15716)

Summary:

Changes:
1. add `--seq_len` param to llama script to distinguish max_seq_len which is compile time param
2. Add warnings in the runner when `seq_len` is clamped to `max_seq_len` to avoid silently clamping it.
3. Add warnings in the token generator when EOS is not reached due to insufficient seq_len or max_seq_len.

Differential Revision: D86696759

winskuo-quic reviewed (approved):
Thanks for the explanation. 
I am slightly leaning toward leaving warnings messages in runtime and reuse the `--max_seq_len` flag in `llama.py`, which aligns with your new commit.
Thanks for the help on improving user's experience!

cccclai commented:
Let's rename the PR title because it's not introducing seq_len anymore
